### PR TITLE
feat(macrobenchmarks): add dataloading knobs to HF workload

### PR DESCRIPTION
**Pull request by zhixiangli**
- Add epochs, shuffle buffer size, and prefetch factor configuration knobs.
- Propagate these knobs from Cloud Build through Helm to the workload.
- Use StatefulDataLoader from torchdata to support resuming from checkpoints.
- Use DatasetEpochCallback to advance shuffle seed per epoch.
- Record these new configuration parameters in the benchmark metrics schema.

### CODE REVIEW

File: gcsfs/tests/perf/macrobenchmarks/workloads/hf-pytorch-lightning-cpu/helm_chart/llama_3_1_8b_cpu_sim.py [hunk 1]
```diff
@@ -239,6 +255,49 @@ def collate_fn(examples):
...
+        data_files=f"{path}/*.parquet",
+        split="train",
+        streaming=True,
+        columns=["text"],
+    )
+    if shuffle_buffer_size > 0:
+        ds = ds.shuffle(
+            seed=shuffle_seed,
+            buffer_size=shuffle_buffer_size,
+            max_buffer_input_shards=1,
```
Comment (yuxin00j) [inline gcsfs/tests/perf/macrobenchmarks/workloads/hf-pytorch-lightning-cpu/helm_chart/llama_3_1_8b_cpu_sim.py]: should we make this configurable?

File: gcsfs/tests/perf/macrobenchmarks/workloads/hf-pytorch-lightning-cpu/helm_chart/llama_3_1_8b_cpu_sim.py [hunk 2]
```diff
@@ -383,6 +442,17 @@ def configure_optimizers(self):
...
 # ---- Callbacks ------------------------------------------------------------
+class DatasetEpochCallback(Callback):
+    """Advances HF dataset shuffle seed per epoch (workaround for Lightning + IterableDataset)."""
+
+    def __init__(self, dataset):
+        super().__init__()
+        self._dataset = dataset
+
+    def on_train_epoch_start(self, trainer, pl_module):
+        self._dataset.set_epoch(trainer.current_epoch)
```
[the remark this answers is not in the document]
Comment (zhixiangli) [reply gcsfs/tests/perf/macrobenchmarks/workloads/hf-pytorch-lightning-cpu/helm_chart/llama_3_1_8b_cpu_sim.py:485]: Wrong

**Review by yuxin00j — APPROVED** [review]
